### PR TITLE
Multiline input

### PR DESCRIPTION
**Pull request by IanCStewart**

### CODE REVIEW

**Comment by MartijnReeuwijk** [comment]
It works on the desktop version but not on the mobile version

**Comment by sjaakluthart** [comment]
@IanCStewart please fix the things @MartijnReeuwijk mentioned

**Comment by MartijnReeuwijk** [comment]
The styling works we do need to test the send message code so you cant send an empty message

File: src/message-input/index.jsx [hunk 1]
```diff
@@ -3,71 +3,129 @@ import React, { Component } from 'react';
...
+    this.textarea.style.height = '100%';
+
+    onChange(event);
   }
 
-  constructor() {
-    super();
+  handleKeyDown = (event) => {
+    const { multiLine } = this.props;
+    const md = new MobileDetect(window.navigator.userAgent);
```
Comment (sjaakluthart) [inline src/message-input/index.jsx]: Move this declaration to top of file

**Comment by sjaakluthart** [comment]
Add maxRows prop

File: src/input/index.jsx [hunk 2]
```diff
@@ -76,31 +31,147 @@ Input.propTypes = {
...
+      error,
+      errorStyle,
+      placeholder,
+      placeholderStyle,
+      multiLine,
+      maxRows,
+      ...custom
+    } = this.props;
+    const { height } = this.state;
+    const { lineHeight } = getStyles.textarea(error, inputStyle);
```
Comment (sjaakluthart) [inline src/input/index.jsx]: Don't call `getStyles.textarea` twice, save the first result in a constant

File: src/input/index.jsx [hunk 3]
```diff
@@ -76,31 +31,147 @@ Input.propTypes = {
...
+      );
+    }
+
+    return (
+      <section style={getStyles.root(disabled, style)}>
+        <label style={getStyles.label(labelStyle)} htmlFor={name}>{label}</label>
+        {input}
+        <Style
+          rules={{
+            '.input::placeholder': combineStyles({ color: colors.placeholderText }, placeholderStyle)
```
Comment (sjaakluthart) [inline src/input/index.jsx]: Add a getStyles function for placeholder

File: src/internal/mobile-detect.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import MobileDetect from 'mobile-detect';
+
+export default new MobileDetect(window.navigator.userAgent);
```
Comment (sjaakluthart) [inline src/internal/mobile-detect.js]: Make sure window, window.navigator and window.navigator.userAgent are defined... This will break component tests since mocha does not have the window object

File: src/message-input/index.jsx [hunk 5]
```diff
@@ -93,40 +164,74 @@ class MessageInput extends Component {
...
       rightButton,
+      multiLine,
+      maxRows,
       sendIconColor,
       sendIcon,
       placeholderStyle,
       ...custom
     } = this.props;
+    const { height } = this.state;
+    const { lineHeight } = getStyles.textarea(inputStyle);
```
Comment (sjaakluthart) [inline src/message-input/index.jsx]: Don't call getStyles.textarea twice

File: src/input/index.jsx [hunk 6]
```diff
@@ -76,31 +29,148 @@ Input.propTypes = {
...
+      />
+    );
+
+    if (multiLine) {
+      input = (
+        <section style={getStyles.inputRoot(height)}>
+          <textarea
+            className="input"
+            style={textareaStyle}
+            onChange={event => this.handleChange(event, maxRows, parseInt(lineHeight, 10))}
```
Comment (sjaakluthart) [inline src/input/index.jsx]: What happens if lineHeight is not a parseable string or specified in pixels? This can occur if a consumer passes style you didn't consider, something like `lineHeight: '2em'`...
